> > The testing can be performed as shown:
> > 
> > * build the docker image: `docker build --progress=plain -t rv32emu-slim .`
> > * spin up a container: `docker run -it rv32emu-slim`
> > * execute a elf file: `rv32emu ./build/hello.elf`
> 
> Due to Docker being cumbersome and resource-intensive on macOS with Apple Silicon, I prefer using [nerdctl](https://github.com/containerd/nerdctl), which offers a command line interface that is compatible with Docker.
> 
> Along with [lima](https://github.com/lima-vm/lima), I run the command `nerdctl.lima build --progress=plain -t rv32emu-slim .` and the following message appears:
> 
> ```
> #11 [base_gcc 6/6] RUN make tool
> #11 0.144   CC	build/rv_histogram.o
> #11 0.232   LD	build/rv_histogram
> #11 0.240 build/riscv.o: file not recognized: file format not recognized
> #11 0.240 collect2: error: ld returned 1 exit status
> #11 0.241 make: *** [mk/tools.mk:24: build/rv_histogram] Error 1
> #11 ERROR: process "/bin/sh -c make tool" did not complete successfully: exit code: 2
> ```

Things are getting interesting, as I can't reproduce this error.

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,38 +1,24 @@
-FROM sysprog21/rv32emu-sail as base_sail
+FROM alpine:3.19 as base_gcc
 
-FROM ubuntu:22.04 as base_gcc
-
-# install extra packages for the emulator to compile with full capabilities correctly
-RUN apt-get update && \
-    DEBIAN_FRONTEND=noninteractive apt-get install -y \
-    build-essential curl git && \
-    rm -rf /var/lib/apt/lists/*
+RUN apk add --update alpine-sdk git curl
 
 # copy in the source code
 WORKDIR /home/root/rv32emu
 COPY . .
 
 # generate execution file for rv32emu and rv_histogram
-RUN make
+RUN make ENABLE_SDL=0
 RUN make tool
 
-FROM ubuntu:22.04 as final
+FROM alpine:3.19 as final
 
-# set up the timezone
-ENV TZ=Asia/Taipei
-RUN ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone
-
-# copy in elf files and reference emulator
-WORKDIR /home/root/rv32emu
+# copy in elf files
 COPY ./build/*.elf /home/root/rv32emu/build/
-COPY ./tests/arch-test-target/sail_cSim/riscv_sim_RV32 /home/root/rv32emu/tests/arch-test-target/sail_cSim/
-
-# replace the emulator (riscv_sim_RV32) with the arch that the container can execute 
-RUN rm /home/root/rv32emu/tests/arch-test-target/sail_cSim/riscv_sim_RV32
-COPY --from=base_sail /home/root/riscv_sim_RV32 /home/root/rv32emu/tests/arch-test-target/sail_cSim/
 
 # get rv32emu and rv_histogram binaries
 COPY --from=base_gcc /home/root/rv32emu/build/rv32emu /home/root/rv32emu/build/rv32emu
 COPY --from=base_gcc /home/root/rv32emu/build/rv_histogram /home/root/rv32emu/build/rv_histogram
 
-ENV PATH=/home/root/rv32emu/build:/home/root/rv32emu/tests/arch-test-target/sail_cSim:$PATH
+ENV PATH=/home/root/rv32emu/build:$PATH
+
+WORKDIR /home/root/rv32emu

